src/main/java/com/exasol/adapter/dynamodb/documentpath/DocumentPathWalker.java

```diff
@@ -0,0 +1,129 @@
+package com.exasol.adapter.dynamodb.documentpath;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import com.exasol.adapter.dynamodb.documentnode.DocumentArray;
+import com.exasol.adapter.dynamodb.documentnode.DocumentNode;
+import com.exasol.adapter.dynamodb.documentnode.DocumentObject;
+
+/**
+ * This class walks a given path defined in {@link DocumentPathExpression} through a {@link DocumentNode} structure.
```

Comment:
I still don't get from the name what exactly Walker does

Reply:
is it more understandable like this?